 Wifi : Ignore QNS SCAN if data traffic is high.

Ignore Quality network selection  SCANs if there is a high data traffic.

Thresholds are as follows.
txSuccessRate > MAX_TX_PACKET_FOR_PARTIAL_SCANS(40)
rxSuccessRate > MAX_RX_PACKET_FOR_PARTIAL_SCANS(80)

diff --git a/service/java/com/android/server/wifi/WifiConnectivityManager.java b/service/java/com/android/server/wifi/WifiConnectivityManager.java
index 5b285f5..b0ea780 100644
--- a/service/java/com/android/server/wifi/WifiConnectivityManager.java
+++ b/service/java/com/android/server/wifi/WifiConnectivityManager.java
@@ -782,7 +782,17 @@ private void startPeriodicSingleScan() {
         if (mStateMachine.getScanCount()  < mStateMachine.getMaxConfiguredScanCount()) {
             isFullBandScan = false;
         }
-        startSingleScan(isFullBandScan);
+        if (mWifiState == WIFI_STATE_CONNECTED
+                && (mWifiInfo.txSuccessRate
+                            > mConfigManager.MAX_TX_PACKET_FOR_PARTIAL_SCANS
+                    || mWifiInfo.rxSuccessRate
+                            > mConfigManager.MAX_RX_PACKET_FOR_PARTIAL_SCANS)) {
+            localLog("Ignore scan due to heavy traffic, txSuccessRate="
+                        + mWifiInfo.txSuccessRate + " rxSuccessRate="
+                        + mWifiInfo.rxSuccessRate);
+        } else {
+            startSingleScan(isFullBandScan);
+        }
         schedulePeriodicScanTimer(mPeriodicSingleScanInterval);
 
         // Set up the next scan interval in an exponential backoff fashion.
